src/FileHandler.java

```diff
@@ -0,0 +1,55 @@
+import java.io.*;
+import java.util.ArrayList;
+
+public class FileHandler {
+
+    public static ArrayList<String> readFileToArrayList(String fileName) throws IOException {
+        File file = new File(fileName);
+        ArrayList<String> calculatedValues = new ArrayList<>();
+
+        try (
+                var fileReader = new FileReader(fileName);
+                var reader = new BufferedReader(fileReader)
+        ) {
+            String nextLine;
+            while ((nextLine = reader.readLine()) != null) {
+                calculatedValues.add(processLine(nextLine));
+            }
+            return calculatedValues;
+        }
+    }
+
+
+    public static void saveResultToFile(ArrayList<String> result, String fileName) {
+        try (
+                var fileWriter = new FileWriter(fileName);
+                var writer = new BufferedWriter(fileWriter)
+        ) {
+            for (String s : result) {
+                writer.write(s);
+                writer.newLine();
+            }
+
+            System.out.println("Report created");
+        } catch (IOException e) {
+            System.err.println("Unable to save the file " + fileName);
+        }
+    }
+
+    private static String processLine(String nextLine) {
+        String[] splitValues;
+        if (nextLine.contains("+")) {
```

Comment:
lepiej rozdzielic najpierw linijke na odzielne pola(leftTerm, operator, righTerm) i zrobic switcha dla wartosci operator

Reply:
trudno będzie to sensownie robić to rozdzielenie żeby było mniej kodu , bo trzeba by to po jakiejś spacji parsować, a wtedy przestaną działać wyrażenia bez spacji typu 3*9, albo mieszane typu 3* 9 z białymi spacjami wydaje mi nie, że zrobiłem to optymalnie aczkolwiek mogłem przejść całą linię po wykonaniu na niej trim() w poszukiwaniu jednego z 3 znaków działań. W każdym razie ilość kodu znacznie by urosła. Ale to już ocena subiektywna. Nie będę tego poprawiać, ale rozumiem co miałeś na myśli.